src/txn/txn_recover.c

```diff
@@ -1012,6 +1012,9 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
          */
         WT_ERR(session->iface.checkpoint(&session->iface, "force=1"));
 
+    /* Remove any backup file now that metadata has been synced. */
+    WT_ERR(__wt_backup_file_remove(session));
```

Comment:
Do we need to wait this long to remove the backup file? Would it be sufficient to remove it after we have built a new metadata file (after `metadata_hot_load_backup()`, I think)?

I'm specifically concerned that a by waiting until after the recovery checkpoint, there is a risk that we'll wind up removing the checkpoint backup depends on and, possibly, overwriting some of it's blocks. I don't think that can happen right now because we don't do any IO between the checkpoint and the file remove. But I would worry that a future change might make that possible and then bite us here.

Reply:
If we wanted to remove the backup file earlier, we would need to make sure the metadata is reconciled and fsync-ed.  Currently that mechanism happens as part of a system wide checkpoint.  `__wt_checkpoint_close()` does have the ability to checkpoint a single file (via `__checkpoint_tree`), would we need to do something like that?

Reply:
The removal was right after `metadata_hot_load_backup` originally. In the recovery checkpoint, we know whether we were a backup, I think it would be cleaner and safer to not delete any checkpoints in that case.